src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -454,6 +464,12 @@ private async Task LoadIntoBufferAsyncCore(Task serializeToStreamTask, MemoryStr
             }
         }
 
+        protected virtual Stream CreateContentReadStream(CancellationToken cancellationToken)
```

Comment:
Won't we want to override this on most of our HttpContent-derived instances, just as we do with CreateContentReadStreamAsync?

Reply:
Yes. Is something wrong with it? Did I miss something?

Reply:
> Is something wrong with it? Did I miss something?

We're not overriding it in this PR.

Reply:
Ahh, I see. I guess we should. 
The thing is we don't actually have to, because I'm calling [`TryReadAsStream`](https://github.com/dotnet/runtime/blob/9ea053a5d739a2ab2b6ba456a875ee9feefc15c2/src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs#L264) first (in `ReadAsStream`) which ends up in `TryCreateContentReadStream` which we already override for all our `HttpContent`s.

Maybe the call to `TryReadAsStream` is wrong? Maybe we don't actually need virtual `CreateContentReadStream`? Maybe we have to override `CreateContentReadStream` anyway, because I missed some use-case?

That's why I put up this PR, I'm not sure about it.

Reply:
> The thing is we don't actually have to, because I'm calling TryReadAsStream first (in ReadAsStream) which ends up in TryCreateContentReadStream which we already override for all our HttpContents.

We don't override it on all of our contents, only on those that can synchronously / without any I/O produce the Stream, e.g. we don't override it on MultipartContent.

I don't think the call to TryReadFromStream is necessary here.  We can just call CreateContentReadStream, appropriately overridden on all derived types.  The internal TryReadFromStream exists purely for use in async code paths where we can utilize a fast-path to get a cheaply-created Stream and without having to incur the additional Task overhead.